apps/web/src/lib/middleware/next.ts

```diff
@@ -0,0 +1,100 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { NextMiddlewareChain, NextMiddlewareFunction } from './types';
+import { CONNECT_SOURCES } from '../../config/connect-sources';
+import { IMAGE_HOSTS } from '../../config/image-hosts';
+
+/**
+ * Composes multiple Next.js middleware functions into a single middleware function
+ * @param middlewareChain Array of middleware functions to compose
+ * @returns A single middleware function that executes the chain
+ */
+export function composeMiddleware(
+  middlewareChain: NextMiddlewareChain,
+): NextMiddlewareFunction {
+  return async (request: NextRequest, response?: NextResponse) => {
+    const context = { request, response };
+
+    for (const middleware of middlewareChain) {
+      const result = await middleware(context.request, context.response);
+      if (result) {
+        context.response = result;
+      }
+
+      // If context.response is set and it has a redirect, return early
+      if (
+        context.response &&
+        (context.response.headers.get('Location') ||
+          context.response.status === 301 ||
+          context.response.status === 302)
+      ) {
+        return context.response;
+      }
+    }
+
+    return context.response;
+  };
+}
+
+/**
+ * Content security policy (CSP) middleware
+ * @returns Middleware function
+ */
+export function cspMiddleware(): NextMiddlewareFunction {
+  // FIXME: workaround to meet CSP. These two scripts ignore nonce
+  const SCRIPT_HASHES = [
+    // Uploadthing
+    "'sha256-9rh1hg0t8gzBb+71sg04fUOw1ZnwOMplqN4Jqi1j5o4='",
+    // Next theme
+    "'sha256-n46vPwSWuMC0W703pBofImv82Z26xo4LXymv0E9caPk='",
+  ].join(' ');
+  const imageSrc = [
+    'data:',
+    ...IMAGE_HOSTS.map((host) => `https://${host}`),
+  ].join(' ');
+  const connectSrc = [
+    ...CONNECT_SOURCES,
+    process.env.NEXT_PUBLIC_RPC_URL ?? '',
+  ].join(' ');
+
+  return async (request: NextRequest) => {
+    const nonce = Buffer.from(crypto.randomUUID()).toString('base64');
+    // FIXME: enable nonce
+    const unsafeForDevelopment = "'unsafe-inline' 'unsafe-eval'";
```

Comment:
Let's just return early in development mode without returning any config. There is no need for CSP at all when developing locally.

Reply:
Please also clean up the commented code when ready.